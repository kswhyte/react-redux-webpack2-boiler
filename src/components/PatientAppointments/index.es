import Component from './PatientAppointments';
export default Component;
