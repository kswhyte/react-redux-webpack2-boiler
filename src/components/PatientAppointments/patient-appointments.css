@import "imports/_helpers.css";

.patient-appointments-wrapper {
    & .appt-tables {
        margin-top: 40px;
        width: 100%;
    }
    & .header-table th {
        border-bottom: solid 3px var(--light-grey);
    }
    & .appt-tables th {
        padding: 0px 20px 5px;
        font-family: Open Sans;
        font-size: 12px;
        font-weight: bold;
    }


    & .appt-tables td,th {
        width: (100%/6)%;
    }
    & .appt-tables td {
        font-size: 12px;
        padding: 20px 10px 20px;
    }
    & .appt-tables:not(:first-child) tr {
        border-bottom: solid 2px var(--light-grey);
    }

}

.upcoming-appt-head {
    font-family: Open Sans;
    font-weight: bold;
    font-size: 10px;
    color: var(--future-green);
}
.past-appt-head {
    font-family: Open Sans;
    font-weight: bold;
    font-size: 10px;
    color: var(--dull-orange);
}