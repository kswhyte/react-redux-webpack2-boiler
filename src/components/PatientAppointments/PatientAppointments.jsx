import React from 'react';
import PropTypes from 'prop-types';
import Moment from 'moment';

import './patient-appointments.css';

const PatientAppointments = props => {
  PatientAppointments.propTypes = {
    children: PropTypes.object,
    appointments: PropTypes.array.required
  };

  const pastAppts = props.appointments.filter(appt => {
    return Moment(appt.appointmentDateTime) < Moment();
  });

  const upcomingAppts = props.appointments.filter(appt => {
    return Moment(appt.appointmentDateTime) > Moment();
  });

  return (
    <div className="patient-appointments-wrapper">
      <div className="tab-content clearfix">
        <div className="patient-appointments-form tab-pane active">
          {props.appointments.length === 0 &&
            <div className="page-instructions">
              <p>Patient has no appointment history</p>
            </div>}
          <table className="appt-tables header-table">
            <thead>
              <th>Date</th>
              <th>Time</th>
              <th>Provider Name</th>
              <th>Phone</th>
              <th>Scheduler</th>
              <th>Actions</th>
            </thead>
          </table>
          {upcomingAppts.length > 0 && <UpcomingAppointments upcomingAppts={upcomingAppts} />}
          {pastAppts.length > 0 && <PastAppointments pastAppts={pastAppts} />}
        </div>
      </div>
    </div>
  );
};

const PastAppointments = props => {
  PastAppointments.propTypes = {
    pastAppts: PropTypes.array.required
  };
  return (
    <table className="appt-tables">
      <thead>
        <th colSpan={6} className="past-appt-head">
          Past Appointments
        </th>
      </thead>
      <tbody>
        {props.pastAppts.map(appt => {
          return (
            <tr key={appt.appointmentID} >
              <td>{Moment(appt.appointmentDateTime).format('LL')}</td>
              <td>{Moment(appt.appointmentDateTime).format('LT')}</td>
              <td>{appt.providerName}</td>
              <td>{appt.officePhone}</td>
              <td>{appt.agentName}</td>
              <td></td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

const UpcomingAppointments = props => {
  UpcomingAppointments.propTypes = {
    upcomingAppts: PropTypes.array.required
  };
  return (
    <table className="appt-tables">
      <thead>
        <th colSpan={6} className="upcoming-appt-head">
          Upcoming Appointments
        </th>
      </thead>
      <tbody>
        {props.upcomingAppts.map(appt => {
          return (
            <tr key={appt.appointmentID} >
              <td>{Moment(appt.appointmentDateTime).format('LL')}</td>
              <td>{Moment(appt.appointmentDateTime).format('LT')}</td>
              <td>{appt.providerName}</td>
              <td>{appt.officePhone}</td>
              <td>{appt.agentName}</td>
              <td><button>CANCEL</button></td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export default PatientAppointments;
