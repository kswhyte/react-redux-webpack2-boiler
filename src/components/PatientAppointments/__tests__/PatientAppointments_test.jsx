import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import PatientAppointments from '../PatientAppointments';
import Moment from 'moment';
// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { mount } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('PatientAppointments', module);

Date.now = jest.fn(() => 1487076708000);

stories.add('With Appointments', () => {


  const props = {
    appointments: [{
      appointmentID: "12345",
      appointmentDateTime: Moment().add(7,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },
    {
      appointmentID: "12346",
      appointmentDateTime: Moment().add(8,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },
    {
      appointmentID: "12347",
      appointmentDateTime: Moment().add(9,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },{
      appointmentID: "12348",
      appointmentDateTime: Moment().subtract(7,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },{
      appointmentID: "12349",
      appointmentDateTime: Moment().subtract(6,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    }]
  }
  const patientAppointmentsStory = <PatientAppointments {...props} />;

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should render the PatientAppointments component without crashing', () => {
        let output = mount(patientAppointmentsStory);
        expect(output.find('.patient-appointments-wrapper').length).toEqual(1);
      });
      it('Should show tables as expected', () => {
        let output = mount(patientAppointmentsStory);
        expect(output.find('td').length).toEqual(30);
        expect(output.find('th').length).toEqual(8);
      });

    })
  );
  return patientAppointmentsStory;
});

stories.add('With only Past Appointments', () => {

  const propsNoAppts = {
    appointments: [{
      appointmentID: "12348",
      appointmentDateTime: Moment().subtract(6,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },{
      appointmentID: "12349",
      appointmentDateTime: Moment().subtract(7,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    }]
  }

  const patientAppointmentsStory = <PatientAppointments {...propsNoAppts} />;
  specs(() =>
    describe('Show a successful alert', () => {
      it('Should show only past appointments', () => {
        let output = mount(patientAppointmentsStory);
        expect(output.find('td').length).toEqual(12);
      });
    })
  );

  return patientAppointmentsStory;
});

stories.add('With No Appointments', () => {

  const propsNoAppts = {
    appointments: []
  }

  const patientAppointmentsStory = <PatientAppointments {...propsNoAppts} />;

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should show message if no appointments found', () => {
        let output = mount(patientAppointmentsStory);
        expect(output.find('td').length).toEqual(0);
        expect(output.find('.page-instructions > p').text()).toContain("Patient has no appointment history");
      });
    })
  );
  return patientAppointmentsStory;
});

stories.add('With Just Future Appointments', () => {

  const propsFutureAppts = {
    appointments: [{
      appointmentID: "12345",
      appointmentDateTime: Moment().add(7,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },
    {
      appointmentID: "12346",
      appointmentDateTime: Moment().add(8,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    },
    {
      appointmentID: "12347",
      appointmentDateTime: Moment().add(9,'d').format(),
      providerName: "Dr Name Name",
      agentName: "Name",
      officePhone: "123-456-7891"
    }]
  }

  const patientAppointmentsStory = <PatientAppointments {...propsFutureAppts} />;

  specs(() =>
    describe('Show a successful alert', () => {
      it('Should show Only future page', () => {
        let output = mount(<PatientAppointments {...propsFutureAppts} />);
        expect(output.find('td').length).toEqual(18);
      });
    })
  );
  return patientAppointmentsStory;
});
