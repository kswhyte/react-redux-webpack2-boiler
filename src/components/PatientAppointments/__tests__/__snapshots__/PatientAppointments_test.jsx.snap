// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story 1`] = `
<div
  className="patient-appointments-wrapper"
>
  <div
    className="tab-content clearfix"
  >
    <div
      className="patient-appointments-form tab-pane active"
    >
      <form>
        <div>
          <div
            className="col-sm-2"
          >
            <div />
          </div>
          <div
            className="col-sm-1"
          >
            <div
              label="Time"
            />
          </div>
        </div>
        <div
          className="line"
        />
        <h3>
          CREATE NEW APPOINTMENT
        </h3>
        <h3>
          PAST APPOINTMENTS
        </h3>
      </form>
    </div>
  </div>
</div>
`;
